scripts/azureml-assets/azureml/assets/release/asset_publish.py

```diff
@@ -52,6 +52,45 @@ def test_files_preprocess(test_jobs, asset_ids: dict):
                     default_flow_style=False,
                     sort_keys=False)
 
+def update_spec_file(spec_file: Path, path: Path):
+    "update the yaml file after getting the model has been prepared"
+    with open(spec_file) as f:
+        model_file = yaml.safe_load(f)
+    model_file['path'] = path
+    with open(spec_file, 'w') as f:
+        yaml.dump(model_file, f)
+
+def model_prepare(model_config : assets.ModelConfig, spec_file: Path) -> str :
+    """
+    Prepares the model. Downloads the model if required and converts the models to specified
+    publish type.
+
+    Return: returns the local path to the model.
+    """
+    if (model_config.path_local_path):
+        return None
+
+    model_dir = '/tmp/' + model_config.name
```

Comment:
not sure if this is handled, agent storage will not be sufficient for models we are going to download.
@wayliums @elesel can you please share specs for agents we are using currently for component and environment. I propose using a different pool of agents for model publishing

@skasturi

Reply:
Please ask this question via Teams and we'll discuss there.

Reply:
Moving this conversation to teams